woocommerce/woocommerce-android
# Issue/3207 add attribute part3

This PR continues the work on #3207  by supporting adding new attributes and options, drag-and-drop re-ordering of options, removal of options, and saving attribute changes to the backend.

To test:

- Go to product detail for a variable product which has variations
- Tap Variations
- In the variation list, tap the options menu and select "Edit attributes"
- Tap "Add Attribute"
- Enter a new attribute name to be taken to the options (terms) screen 
- Add a few options
- Return to product detail and ensure the Update button is showing and tapping it saves the attribute changes
- Return to the attribute options screen
- Drag-and-drop to re-order options, tap "X" to delete an option
- Return again to product detail and ensure changes are saved
- View the product on the web and ensure changes are reflected

**Note:** This PR focuses on local terms almost entirely. There's still a lot of work to do to support global terms.

![terms](https://user-images.githubusercontent.com/3903757/109699351-77ae8380-7b5e-11eb-808d-96f6b68ef8f1.gif)

Update release notes:
